data_collection/gazette/spiders/rj_duque_de_caxias.py

```diff
@@ -0,0 +1,128 @@
+import re
+from datetime import date, datetime
+
+import scrapy
+from dateparser import parse
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RjDuqueDeCaxiasSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3301702"
+    name = "rj_duque_de_caxias"
+    allowed_domains = ["duquedecaxias.rj.gov.br"]
+    portal_url = "https://duquedecaxias.rj.gov.br/portal/{}.html"
+    directory_url = "https://duquedecaxias.rj.gov.br/portal/boletim-oficial/{}/{}"
+
+    months = [
+        "01-Janeiro",
+        "02-Fevereiro",
+        "03-Marco",
+        "04-Abril",
+        "05-Maio",
+        "06-Junho",
+        "07-Julho",
+        "08-Agosto",
+        "09-Setembro",
+        "10-Outubro",
+        "11-Novembro",
+        "12-Dezembro",
+    ]
+
+    def start_requests(self):
```

Comment:
We need to support `start_date` and `end_date`. With the current approach is only possible to crawl all gazettes. Could you please add support for it?

Reply:
Ok @anapaulagomes,  no problem! But for the years below 2017 I can't consider the days to filter, i have no way to get them, this is a problem?

Reply:
Oh I see! What do you think of displaying a warning for this case and only filter by year? If the filtering is too much trouble, a warning could be just fine.

Reply:
I can filter by months too. Only daily filtering is the problem.

How can I display this warning, are you have some example, please 😖?

Reply:
Oh even better! :)

You can do it like this:

```
import warnings

warnings.warn("It isn't possible to filter by day. Using month and year.")
```
https://docs.python.org/3/library/warnings.html

Reply:
Ok thanks @anapaulagomes